### ISSUE TITLE
System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.1.0.0' after adding BenchmarkDotNet.Diagnostics.Windows

### ISSUE THREAD

**Issue by mattwarren** [dotnet/BenchmarkDotNet#942]
I created a new benchmark project (.NET Framework 4.6), added [BenchmarkDotNet 0.11.2](https://www.nuget.org/packages/BenchmarkDotNet/) and all was fine. However as soon as I added [BenchmarkDotNet.Diagnostics.Windows 0.11.2](https://www.nuget.org/packages/BenchmarkDotNet.Diagnostics.Windows/) I started getting this error:

```
Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   --- End of inner exception stack trace ---
   at BenchmarkDotNet.Running.BenchmarkConverter.<>c__DisplayClass8_0.<GetParameterDefinitions>g__GetDefinitions|0[TAttribute](Func`3 getValidValues)
   at BenchmarkDotNet.Running.BenchmarkConverter.GetParameterDefinitions(Type type)
   at BenchmarkDotNet.Running.BenchmarkConverter.MethodsToBenchmarksWithFullConfig(Type containingType, MethodInfo[] benchmarkMethods, ReadOnlyConfig fullConfig)
   at BenchmarkDotNet.Running.BenchmarkConverter.TypeToBenchmarks(Type type, IConfig config)
   at BenchmarkDotNet.Running.BenchmarkRunner.Run[T](IConfig config)
   at AgentBenchmarks.Program.Main(String[] args) in C:\Users\MattW\Dev\APM\AgentBenchmarks\AgentBenchmarks\Program.cs:line 16
```

Not sure why it's failing twice, for `System.Runtime, Version=4.1.0.0` and `System.Runtime, Version=4.0.0.0`.

My project has a reference to `System.Runtime 4.3.0` (added with the BenchmarkDotNet NuGet pkgs), I tried to 'downgrade' that to `System.Runtime 4.1.0`, but then I get this message:

![image](https://user-images.githubusercontent.com/157298/47859602-9497e800-dde6-11e8-88b9-4539216350e7.png)

I also tried adding a direct reference to `System.Runtime.dll` (version 4.1.0), but then I get these errors in the benchmark output:

```
// Build Exception: The type initializer for 'Roslyn.Utilities.StringTable' threw an exception.
```

Am I doing something wrong?

I really want to try out the [EtwProfiler](https://benchmarkdotnet.org/articles/features/etwprofiler.html) because I think it's a great feature! Thanks @adamsitnik

**Comment by adamsitnik** [comment]
@mattwarren  did VS generate some App.config file with binding redirects? if so, does removing it helps?

**Comment by mattwarren** [comment]
Yeah VS did generate a load of bindings redirects in App.config. But
unfortunately removing them doesn't help?

**Comment by adamsitnik** [comment]
Hmm...

Could you try to create new project and install only `BenchmarkDotNet.Diagnostics.Windows`? It has a dependency to `BenchmarkDotNet` to maybe VS is going to generate the right ones...

If this does not help, try net471 or upper (it might resolve different  .NET Standard assemblies)

**Comment by mattwarren** [comment]
> If this does not help, try net471 or upper (it might resolve different .NET Standard assemblies)

Yeah, that did the trick, thanks! (for me, moving the project to .NET Framework 4.7.2 isn't a problem)

**Comment by mattwarren** [comment]
Although I don't know where this leaves people who want to use `BenchmarkDotNet.Diagnostics.Windows` with .NET 4.6? Is moving to 4.7.1 the only option?

> Could you try to create new project and install only `BenchmarkDotNet.Diagnostics.Windows`? It has a dependency to BenchmarkDotNet to maybe VS is going to generate the right ones...

Wierd, I tried that, i,e, empty project, just adding `BenchmarkDotNet.Diagnostics.Windows` and I get this amazingly cryptic error!!

```
Unhandled Exception:
   Cannot print exception string because Exception.ToString() failed.
```

**Comment by AndreyAkinshin** [comment]
If we create an SDK-style project and change `TargetFramework` to `net46`, everything works correctly. It can be used as a temporary workaround for .NET 4.6 projects.
However, we still should investigate this problem with classic projects.

**Comment by adamsitnik** [comment]
This was one big PITA, fixed by using some ugly workarounds in #1032
